Configuration configuration
Block[] block_properties